---
title: "DEXSeq_AF22"
author: "Marika Oksanen"
date: "6/30/2021"
output: html_document
---

##DEXSeq

```{r}
library(DEXSeq)
library(biomaRt)
library(VennDiagram)
library(tidyverse)
```

## Preparations

Find the countfiles that are calculated using python scripts
```{r}
countFiles = list.files("/Users/k6-c02dm8d6ml7h/Desktop/Splicing/CountFiles/AF22", pattern=".txt$", full.names=TRUE)
```

```{r}
flattenedFile = list.files("/Users/k6-c02dm8d6ml7h/Desktop/Splicing", pattern="gff$", full.names=TRUE)
```

```{r}
sample_info <- read.delim("~/Desktop//K.Tammimies_20_01_sample_info.txt", header=TRUE)
```

```{r}
sample_info <- separate(sample_info, "User.ID", into = c("sample_number", "cell_line", "treatment"), sep = "_", remove = FALSE )
sample_info <- sample_info[1:43,]
```


```{r}
source("BPA_F.R")
source("Eth_F.R")
source("FH_F.R")
source("Pb_F.R")
source("VPA_F.R")
source("Zn_F.R")
```

```{r}
sample_info$Pb <- Pb_F(sample_info)
sample_info$VPA <- VPA_F(sample_info)
sample_info$BPA <- BPA_F(sample_info)
sample_info$Eth <- Eth_F(sample_info)
sample_info$FH <- FH_F(sample_info)
sample_info$Zn <- Zn_F(sample_info)
```


#### Pb Treatment

```{r}
sample_info <- sample_info[sample_info[, "cell_line"] == "AF22",]
names(sample_info)[names(sample_info) == "Pb"] <- "condition"
```


```{r}
dxd_AF22_Pb = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_AF22_Pb <- estimateSizeFactors(dxd_AF22_Pb)
dxd_AF22_Pb <- estimateDispersions(dxd_AF22_Pb)
plotDispEsts(dxd_AF22_Pb)
```


```{r}
dxr_AF22_Pb = DEXSeq(dxd_AF22_Pb)
```

MA plot:
```{r}
plotMA(dxr_AF22_Pb)
```


Collect significant exons/genes into a data frame:
```{r}
dxr_AF22_Pb_sig <- as.data.frame(dxr_AF22_Pb[dxr_AF22_Pb$padj < 0.05 & !is.na(dxr_AF22_Pb$padj),]) #padj <0.05
dxr_AF22_Pb_sig <- dxr_AF22_Pb_sig[dxr_AF22_Pb_sig$log2fold_treated_control < -0.58 | dxr_AF22_Pb_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_AF22_Pb_sig <- dxr_AF22_Pb_sig[dxr_AF22_Pb_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_AF22_Pb_sig <- dxr_AF22_Pb_sig[order(dxr_AF22_Pb_sig$padj),]
```

```{r}
plotDEXSeq( dxr_AF22_Pb, "ENSG00000153130.17", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE )
```


#### BPA Treatment

```{r}
names(sample_info)[names(sample_info) == "BPA"] <- "condition"
```

```{r}
dxd_AF22_BPA = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_AF22_BPA <- estimateSizeFactors(dxd_AF22_BPA)
dxd_AF22_BPA <- estimateDispersions(dxd_AF22_BPA)
plotDispEsts(dxd_AF22_BPA)
```


```{r}
dxr_AF22_BPA = DEXSeq(dxd_AF22_BPA)
```

MA plot:
```{r}
plotMA(dxr_AF22_BPA)
```


Collect significant exons/genes into a data frame:
```{r}
dxr_AF22_BPA_sig <- as.data.frame(dxr_AF22_BPA[dxr_AF22_BPA$padj < 0.05 & !is.na(dxr_AF22_BPA$padj),]) #padj <0.05
dxr_AF22_BPA_sig <- dxr_AF22_BPA_sig[dxr_AF22_BPA_sig$log2fold_treated_control < -0.58 | dxr_AF22_BPA_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_AF22_BPA_sig <- dxr_AF22_BPA_sig[dxr_AF22_BPA_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_AF22_BPA_sig <- dxr_AF22_BPA_sig[order(dxr_AF22_BPA_sig$padj),]
```

```{r}
plotDEXSeq( dxr_AF22_BPA, "ENSG00000148308.17", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```


#### VPA Treatment

```{r}
names(sample_info)[names(sample_info) == "VPA"] <- "condition"
```

```{r}
dxd_AF22_VPA = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_AF22_VPA <- estimateSizeFactors(dxd_AF22_VPA)
dxd_AF22_VPA <- estimateDispersions(dxd_AF22_VPA)
plotDispEsts(dxd_AF22_VPA)
```


```{r}
dxr_AF22_VPA = DEXSeq(dxd_AF22_VPA)
```

MA plot:
```{r}
plotMA(dxr_AF22_VPA)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_AF22_VPA_sig <- as.data.frame(dxr_AF22_VPA[dxr_AF22_VPA$padj < 0.05 & !is.na(dxr_AF22_VPA$padj),]) #padj <0.05
```

Nothing significant


#### Ethanol Treatment

```{r}
names(sample_info)[names(sample_info) == "Eth"] <- "condition"
```

```{r}
dxd_AF22_Eth = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_AF22_Eth <- estimateSizeFactors(dxd_AF22_Eth)
dxd_AF22_Eth <- estimateDispersions(dxd_AF22_Eth)
plotDispEsts(dxd_AF22_Eth)
```

```{r}
dxr_AF22_Eth = DEXSeq(dxd_AF22_Eth)
```

MA plot:
```{r}
plotMA(dxr_AF22_Eth)
```


Collect significant exons/genes into a data frame:
```{r}
dxr_AF22_Eth_sig <- as.data.frame(dxr_AF22_Eth[dxr_AF22_Eth$padj < 0.05 & !is.na(dxr_AF22_Eth$padj),]) #padj <0.05
dxr_AF22_Eth_sig <- dxr_AF22_Eth_sig[dxr_AF22_Eth_sig$log2fold_treated_control < -0.58 | dxr_AF22_Eth_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_AF22_Eth_sig <- dxr_AF22_Eth_sig[dxr_AF22_Eth_sig$exonBaseMean>= 10,]
```

Nothing significant


#### FH Treatment

```{r}
names(sample_info)[names(sample_info) == "FH"] <- "condition"
```

```{r}
dxd_AF22_FH = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_AF22_FH <- estimateSizeFactors(dxd_AF22_FH)
dxd_AF22_FH <- estimateDispersions(dxd_AF22_FH)
plotDispEsts(dxd_AF22_FH)
```

```{r}
dxr_AF22_FH = DEXSeq(dxd_AF22_FH)
```

MA plot:
```{r}
plotMA(dxr_AF22_FH)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_AF22_FH_sig <- as.data.frame(dxr_AF22_FH[dxr_AF22_FH2$padj < 0.05 & !is.na(dxr_AF22_FH$padj),]) #padj <0.05
dxr_AF22_FH_sig <- dxr_AF22_FH_sig[dxr_AF22_FH_sig$log2fold_treated_control < -0.58 | dxr_AF22_FH_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_AF22_FH_sig <- dxr_AF22_FH_sig[dxr_AF22_FH_sig$exonBaseMean>= 10,]
```

```{r}
plotDEXSeq( dxr_AF22_FH, "ENSG00000102226.9", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```


#### Zn Treatment

```{r}
names(sample_info)[names(sample_info) == "Zn"] <- "condition"
```

```{r}
dxd_AF22_Zn = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```


```{r}
dxd_AF22_Zn <- estimateSizeFactors(dxd_AF22_Zn)
dxd_AF22_Zn <- estimateDispersions(dxd_AF22_Zn)
plotDispEsts(dxd_AF22_Zn)
```

```{r}
dxr_AF22_Zn = DEXSeq(dxd_AF22_Zn)
```

MA plot:
```{r}
plotMA(dxr_AF22_Zn)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_AF22_Zn_sig <- as.data.frame(dxr_AF22_Zn[dxr_AF22_Zn2$padj < 0.05 & !is.na(dxr_AF22_Zn$padj),]) #padj <0.05
dxr_AF22_Zn_sig <- dxr_AF22_Zn_sig[dxr_AF22_Zn_sig$log2fold_treated_control < -0.58 | dxr_AF22_Zn_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
```

Nothing significant

