---
title: "GEeN_ASD17AII"
author: "Marika Oksanen"
date: "7/1/2021"
output: html_document
---

## DEXSeq

```{r}
library(DEXSeq)
library(biomaRt)
library(VennDiagram)
library(tidyverse)
```

Find the countfiles that are calculated using python scripts
```{r}
countFiles = list.files("/Users/k6-c02dm8d6ml7h/Desktop/CountFiles/ASD12/", pattern=".txt$", full.names=TRUE)
```

```{r}
flattenedFile = list.files("/Users/k6-c02dm8d6ml7h/Desktop/", pattern="gff$", full.names=TRUE)
```

```{r}
sample_info <- read.delim("~/Desktop//K.Tammimies_20_01_sample_info.txt", header=TRUE)
```

```{r}
sample_info <- separate(sample_info, "User.ID", into = c("sample_number", "cell_line", "treatment"), sep = "_", remove = FALSE )
sample_info <- sample_info[1:43,]
```

```{r}
source("BPA_F.R")
source("Eth_F.R")
source("FH_F.R")
source("Pb_F.R")
source("VPA_F.R")
source("Zn_F.R")
```

```{r}
sample_info$Pb <- Pb_F(sample_info)
sample_info$VPA <- VPA_F(sample_info)
sample_info$BPA <- BPA_F(sample_info)
sample_info$Eth <- Eth_F(sample_info)
sample_info$FH <- FH_F(sample_info)
sample_info$Zn <- Zn_F(sample_info)
```

#### Pb Treatment

```{r}
names(sample_info)[names(sample_info) == "Pb"] <- "condition"
```

```{r}
dxd_ASD12_Pb = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_ASD12_Pb <- estimateSizeFactors(dxd_ASD12_Pb)
dxd_ASD12_Pb <- estimateDispersions(dxd_ASD12_Pb)
plotDispEsts(dxd_ASD12_Pb)
```


```{r}
dxr_ASD12_Pb = DEXSeq(dxd_ASD12_Pb)
dxr_ASD12_Pbdf <- as.data.frame(dxr_ASD12_Pb)
```

MA plot:
```{r}
plotMA(dxr_ASD12_Pb)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_ASD12_Pb_sig <- as.data.frame(dxr_ASD12_Pb[dxr_ASD12_Pb$padj < 0.05 & !is.na(dxr_ASD12_Pb$padj),]) #padj <0.05
dxr_ASD12_Pb_sig <- dxr_ASD12_Pb_sig[dxr_ASD12_Pb_sig$log2fold_treated_control < -0.58 | dxr_ASD12_Pb_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_ASD12_Pb_sig <- dxr_ASD12_Pb_sig[dxr_ASD12_Pb_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_ASD12_Pb_sig <- dxr_ASD12_Pb_sig[order(dxr_ASD12_Pb_sig$padj),]
```

```{r}
plotDEXSeq( dxr_ASD12_Pb, "ENSG00000256053.7", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```

#### BPA Treatment

```{r}
names(sample_info)[names(sample_info) == "BPA"] <- "condition"
```


```{r}
dxd_ASD12_BPA = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_ASD12_BPA <- estimateSizeFactors(dxd_ASD12_BPA)
dxd_ASD12_BPA <- estimateDispersions(dxd_ASD12_BPA)
plotDispEsts(dxd_ASD12_BPA)
```


```{r}
dxr_ASD12_BPA = DEXSeq(dxd_ASD12_BPA)
dxr_ASD12_BPAdf <- as.data.frame(dxr_ASD12_BPA)
```

MA plot:
```{r}
plotMA(dxr_ASD12_BPA)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_ASD12_BPA_sig <- as.data.frame(dxr_ASD12_BPA[dxr_ASD12_BPA$padj < 0.05 & !is.na(dxr_ASD12_BPA$padj),]) #padj <0.05
dxr_ASD12_BPA_sig <- dxr_ASD12_BPA_sig[dxr_ASD12_BPA_sig$log2fold_treated_control < -0.58 | dxr_ASD12_BPA_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_ASD12_BPA_sig <- dxr_ASD12_BPA_sig[dxr_ASD12_BPA_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_ASD12_BPA_sig <- dxr_ASD12_BPA_sig[order(dxr_ASD12_BPA_sig$padj),]
```

```{r}
plotDEXSeq( dxr_ASD12_BPA, "ENSG00000203804.4", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE, displayTranscripts = TRUE)
```

#### VPA Treatment

```{r}
names(sample_info)[names(sample_info) == "VPA"] <- "condition"
```

```{r}
dxd_ASD12_VPA = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_ASD12_VPA <- estimateSizeFactors(dxd_ASD12_VPA)
dxd_ASD12_VPA <- estimateDispersions(dxd_ASD12_VPA)
plotDispEsts(dxd_ASD12_VPA)
```


```{r}
dxr_ASD12_VPA = DEXSeq(dxd_ASD12_VPA)
dxr_ASD12_VPAdf <- as.data.frame(dxr_ASD12_VPA)
```

MA plot:
```{r}
plotMA(dxr_ASD12_VPA)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_ASD12_VPA_sig <- as.data.frame(dxr_ASD12_VPA[dxr_ASD12_VPA$padj < 0.05 & !is.na(dxr_ASD12_VPA$padj),]) #padj <0.05
dxr_ASD12_VPA_sig <- dxr_ASD12_VPA_sig[dxr_ASD12_VPA_sig$log2fold_treated_control < -0.58 | dxr_ASD12_VPA_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_ASD12_VPA_sig <- dxr_ASD12_VPA_sig[dxr_ASD12_VPA_sig$exonBaseMean>= 10,]
```

NOTHING SIGNIFICANT


#### FH Treatment

```{r}
names(sample_info)[names(sample_info) == "FH"] <- "condition"
```


```{r}
dxd_ASD12_FH = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_ASD12_FH <- estimateSizeFactors(dxd_ASD12_FH)
dxd_ASD12_FH <- estimateDispersions(dxd_ASD12_FH)
plotDispEsts(dxd_ASD12_FH)
```


```{r}
dxr_ASD12_FH = DEXSeq(dxd_ASD12_FH)
dxr_ASD12_FHdf <- as.data.frame(dxr_ASD12_FH)
```

MA plot:
```{r}
plotMA(dxr_ASD12_FH)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_ASD12_FH_sig <- as.data.frame(dxr_ASD12_FH[dxr_ASD12_FH$padj < 0.05 & !is.na(dxr_ASD12_FH$padj),]) #padj <0.05
dxr_ASD12_FH_sig <- dxr_ASD12_FH_sig[dxr_ASD12_FH_sig$log2fold_treated_control < -0.58 | dxr_ASD12_FH_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_ASD12_FH_sig <- dxr_ASD12_FH_sig[dxr_ASD12_FH_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_ASD12_FH_sig <- dxr_ASD12_FH_sig[order(dxr_ASD12_FH_sig$padj),]
```

```{r}
plotDEXSeq( dxr_ASD12_FH, "ENSG00000074201.8", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```

#### Ethanol Treatment

```{r}
names(sample_info)[names(sample_info) == "Eth"] <- "condition"
```


```{r}
dxd_ASD12_Eth = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_ASD12_Eth <- estimateSizeFactors(dxd_ASD12_Eth)
dxd_ASD12_Eth <- estimateDispersions(dxd_ASD12_Eth)
plotDispEsts(dxd_ASD12_Eth)
```


```{r}
dxr_ASD12_Eth = DEXSeq(dxd_ASD12_Eth)
dxr_ASD12_Ethdf <- as.data.frame(dxr_ASD12_Eth)
```

MA plot:
```{r}
plotMA(dxr_ASD12_Eth)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_ASD12_Eth_sig <- as.data.frame(dxr_ASD12_Eth[dxr_ASD12_Eth$padj < 0.05 & !is.na(dxr_ASD12_Eth$padj),]) #padj <0.05
dxr_ASD12_Eth_sig <- dxr_ASD12_Eth_sig[dxr_ASD12_Eth_sig$log2fold_treated_control < -0.58 | dxr_ASD12_Eth_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_ASD12_Eth_sig <- dxr_ASD12_Eth_sig[dxr_ASD12_Eth_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_ASD12_Eth_sig <- dxr_ASD12_Eth_sig[order(dxr_ASD12_Eth_sig$padj),]
```

```{r}
plotDEXSeq( dxr_ASD12_Eth, "ENSG00000126870.15", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE, displayTranscripts = TRUE)
```

#### Zn Treatment

```{r}
names(sample_info)[names(sample_info) == "Zn"] <- "condition"
```

```{r}
dxd_ASD12_Zn = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_ASD12_Zn <- estimateSizeFactors(dxd_ASD12_Zn)
dxd_ASD12_Zn <- estimateDispersions(dxd_ASD12_Zn)
plotDispEsts(dxd_ASD12_Zn)
```


```{r}
dxr_ASD12_Zn = DEXSeq(dxd_ASD12_Zn)
dxr_ASD12_Zndf <- as.data.frame(dxr_ASD12_Zn)
```

MA plot:
```{r}
plotMA(dxr_ASD12_Zn)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_ASD12_Zn_sig <- as.data.frame(dxr_ASD12_Zn[dxr_ASD12_Zn$padj < 0.05 & !is.na(dxr_ASD12_Zn$padj),]) #padj <0.05
dxr_ASD12_Zn_sig <- dxr_ASD12_Zn_sig[dxr_ASD12_Zn_sig$log2fold_treated_control < -0.58 | dxr_ASD12_Zn_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_ASD12_Zn_sig <- dxr_ASD12_Zn_sig[dxr_ASD12_Zn_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_ASD12_Zn_sig <- dxr_ASD12_Zn_sig[order(dxr_ASD12_Zn_sig$padj),]
```

```{r}
plotDEXSeq( dxr_ASD12_Zn, "ENSG00000139618.14", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE, displayTranscripts = TRUE)
```
