---
title: "GeEn_nonrisk_DEU2"
author: "Marika Oksanen"
date: "9/8/2021"
output: html_document
---


```{r}
library(DEXSeq)
library(biomaRt)
library(VennDiagram)
library(tidyverse)
```

Find the countfiles that are calculated using python scripts
```{r}
countFiles = list.files("/Users/k6-c02dm8d6ml7h/Desktop/Splicing/CountFiles/nonrisk", pattern=".txt$", full.names=TRUE)
```

```{r}
flattenedFile = list.files("/Users/k6-c02dm8d6ml7h/Desktop/Splicing", pattern="gff$", full.names=TRUE)
```

```{r}
sample_info <- read.delim("~/Desktop//K.Tammimies_20_01_sample_info.txt", header=TRUE)
sample_info <- separate(sample_info, "User.ID", into = c("sample_number", "cell_line", "treatment"), sep = "_", remove = FALSE )
sample_info <- sample_info[1:43,]
```

```{r}
source("../R_codes/BPA_F.R")
source("../R_codes/Eth_F.R")
source("../R_codes/FH_F.R")
source("../R_codes/Zn_F.R")
source("../R_codes/VPA_F.R")
source("../R_codes/Pb_F.R")
source("../R_codes/risk_F.R")
```

```{r}
sample_info$Pb <- Pb_F(sample_info)
sample_info$VPA <- VPA_F(sample_info)
sample_info$BPA <- BPA_F(sample_info)
sample_info$Eth <- Eth_F(sample_info)
sample_info$FH <- FH_F(sample_info)
sample_info$Zn <- Zn_F(sample_info)
sample_info$risk <- risk_F(sample_info)
```

```{r}
sample_info <- sample_info[-c(32:34),]
```

#### Pb Treatment

```{r}
names(sample_info)[names(sample_info) == "Pb"] <- "condition"
sample_info <- sample_info[sample_info[, "risk"] == "control",]
```


```{r}
formulaFullModel    =  ~ sample + exon + cell_line:exon + condition:exon
formulaReducedModel =  ~ sample + exon + cell_line:exon 
```

```{r}
dxd_nonrisk_Pb = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_nonrisk_Pb <- estimateSizeFactors(dxd_nonrisk_Pb)
dxd_nonrisk_Pb <- estimateDispersions( dxd_nonrisk_Pb, formula = formulaFullModel )
plotDispEsts(dxd_nonrisk_Pb)
```

```{r}
dxd_nonrisk_Pb = testForDEU( dxd_nonrisk_Pb, 
    reducedModel = formulaReducedModel, 
        fullModel = formulaFullModel )
```

```{r}
dxd_nonrisk_Pb <- estimateExonFoldChanges(dxd_nonrisk_Pb)
```

```{r}
dxr_nonrisk_Pb = DEXSeqResults( dxd_nonrisk_Pb )
```

MA plot:
```{r}
plotMA(dxr_nonrisk_Pb)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_nonrisk_Pb_sig <- as.data.frame(dxr_nonrisk_Pb[dxr_nonrisk_Pb$padj < 0.05 & !is.na(dxr_nonrisk_Pb$padj),]) #padj <0.05
dxr_nonrisk_Pb_sig <- dxr_nonrisk_Pb_sig[dxr_nonrisk_Pb_sig$log2fold_treated_control < -0.58 | dxr_nonrisk_Pb_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_nonrisk_Pb_sig <- dxr_nonrisk_Pb_sig[dxr_nonrisk_Pb_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_nonrisk_Pb_sig <- dxr_nonrisk_Pb_sig[order(dxr_nonrisk_Pb_sig$padj),]
```

```{r}
plotDEXSeq( dxr_nonrisk_Pb, "ENSG00000109339.18", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```

#### VPA Treatment

VPA

```{r}
names(sample_info)[names(sample_info) == "VPA"] <- "condition"
sample_info <- sample_info[sample_info[, "risk"] == "control",]
```

```{r}
formulaFullModel    =  ~ sample + exon + cell_line:exon + condition:exon
formulaReducedModel =  ~ sample + exon + cell_line:exon 
```

```{r}
dxd_nonrisk_VPA = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_nonrisk_VPA <- estimateSizeFactors(dxd_nonrisk_VPA)
dxd_nonrisk_VPA <- estimateDispersions( dxd_nonrisk_VPA, formula = formulaFullModel )
plotDispEsts(dxd_nonrisk_VPA)
```

```{r}
dxd_nonrisk_VPA = testForDEU( dxd_nonrisk_VPA, 
    reducedModel = formulaReducedModel, 
        fullModel = formulaFullModel )
```

```{r}
dxd_nonrisk_VPA <- estimateExonFoldChanges(dxd_nonrisk_VPA)
```

```{r}
dxr_nonrisk_VPA = DEXSeqResults( dxd_nonrisk_VPA )
```

MA plot:
```{r}
plotMA(dxr_nonrisk_VPA)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_nonrisk_VPA_sig <- as.data.frame(dxr_nonrisk_VPA[dxr_nonrisk_VPA$padj < 0.05 & !is.na(dxr_nonrisk_VPA$padj),]) #padj <0.05
dxr_nonrisk_VPA_sig <- dxr_nonrisk_VPA_sig[dxr_nonrisk_VPA_sig$log2fold_treated_control < -0.58 | dxr_nonrisk_VPA_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_nonrisk_VPA_sig <- dxr_nonrisk_VPA_sig[dxr_nonrisk_VPA_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_nonrisk_VPA_sig <- dxr_nonrisk_VPA_sig[order(dxr_nonrisk_VPA_sig$padj),]
```

```{r}
plotDEXSeq( dxr_nonrisk_VPA, "ENSG00000087086.13", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```

#### BPA Treatment


```{r}
names(sample_info)[names(sample_info) == "BPA"] <- "condition"
sample_info <- sample_info[sample_info[, "risk"] == "control",]
```

```{r}
formulaFullModel    =  ~ sample + exon + cell_line:exon + condition:exon
formulaReducedModel =  ~ sample + exon + cell_line:exon 
```

```{r}
dxd_nonrisk_BPA = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_nonrisk_BPA <- estimateSizeFactors(dxd_nonrisk_BPA)
dxd_nonrisk_BPA <- estimateDispersions( dxd_nonrisk_BPA, formula = formulaFullModel )
plotDispEsts(dxd_nonrisk_BPA)
```

```{r}
dxd_nonrisk_BPA = testForDEU( dxd_nonrisk_BPA, 
    reducedModel = formulaReducedModel, 
        fullModel = formulaFullModel )
```

```{r}
dxd_nonrisk_BPA <- estimateExonFoldChanges(dxd_nonrisk_BPA)
```

```{r}
dxr_nonrisk_BPA = DEXSeqResults( dxd_nonrisk_BPA )
```

MA plot:
```{r}
plotMA(dxr_nonrisk_BPA)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_nonrisk_BPA_sig <- as.data.frame(dxr_nonrisk_BPA[dxr_nonrisk_BPA$padj < 0.05 & !is.na(dxr_nonrisk_BPA$padj),]) #padj <0.05
dxr_nonrisk_BPA_sig <- dxr_nonrisk_BPA_sig[dxr_nonrisk_BPA_sig$log2fold_treated_control < -0.58 | dxr_nonrisk_BPA_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_nonrisk_BPA_sig <- dxr_nonrisk_BPA_sig[dxr_nonrisk_BPA_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_nonrisk_BPA_sig <- dxr_nonrisk_BPA_sig[order(dxr_nonrisk_BPA_sig$padj),]
```

```{r}
plotDEXSeq( dxr_nonrisk_BPA, "ENSG00000196220.15", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```

#### VPA Treatment

```{r}
names(sample_info)[names(sample_info) == "VPA"] <- "condition"
sample_info <- sample_info[sample_info[, "risk"] == "control",]
```

```{r}
formulaFullModel    =  ~ sample + exon + cell_line:exon + condition:exon
formulaReducedModel =  ~ sample + exon + cell_line:exon 
```

```{r}
dxd_nonrisk_VPA = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_nonrisk_VPA <- estimateSizeFactors(dxd_nonrisk_VPA)
dxd_nonrisk_VPA <- estimateDispersions( dxd_nonrisk_VPA, formula = formulaFullModel )
plotDispEsts(dxd_nonrisk_VPA)
```

```{r}
dxd_nonrisk_VPA = testForDEU( dxd_nonrisk_VPA, 
    reducedModel = formulaReducedModel, 
        fullModel = formulaFullModel )
```

```{r}
dxd_nonrisk_VPA <- estimateExonFoldChanges(dxd_nonrisk_VPA)
```

```{r}
dxr_nonrisk_VPA = DEXSeqResults( dxd_nonrisk_VPA )
```

MA plot:
```{r}
plotMA(dxr_nonrisk_VPA)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_nonrisk_VPA_sig <- as.data.frame(dxr_nonrisk_VPA[dxr_nonrisk_VPA$padj < 0.05 & !is.na(dxr_nonrisk_VPA$padj),]) #padj <0.05
```

Nothing significant

#### FH Treatment

```{r}
names(sample_info)[names(sample_info) == "FH"] <- "condition"
sample_info <- sample_info[sample_info[, "risk"] == "control",]
```

```{r}
formulaFullModel    =  ~ sample + exon + cell_line:exon + condition:exon
formulaReducedModel =  ~ sample + exon + cell_line:exon 
```

```{r}
dxd_nonrisk_FH = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_nonrisk_FH <- estimateSizeFactors(dxd_nonrisk_FH)
dxd_nonrisk_FH <- estimateDispersions( dxd_nonrisk_FH, formula = formulaFullModel )
plotDispEsts(dxd_nonrisk_FH)
```

```{r}
dxd_nonrisk_FH = testForDEU( dxd_nonrisk_FH, 
    reducedModel = formulaReducedModel, 
        fullModel = formulaFullModel )
```

```{r}
dxd_nonrisk_FH <- estimateExonFoldChanges(dxd_nonrisk_FH)
```

```{r}
dxr_nonrisk_FH = DEXSeqResults( dxd_nonrisk_FH )
```

MA plot:
```{r}
plotMA(dxr_nonrisk_FH)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_nonrisk_FH_sig <- as.data.frame(dxr_nonrisk_FH[dxr_nonrisk_FH$padj < 0.05 & !is.na(dxr_nonrisk_FH$padj),]) #padj <0.05
dxr_nonrisk_FH_sig <- dxr_nonrisk_FH_sig[dxr_nonrisk_FH_sig$log2fold_treated_control < -0.58 | dxr_nonrisk_FH_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
```

Nothing significant


#### Zn Treatment

```{r}
names(sample_info)[names(sample_info) == "Zn"] <- "condition"
sample_info <- sample_info[sample_info[, "risk"] == "control",]
```

```{r}
formulaFullModel    =  ~ sample + exon + cell_line:exon + condition:exon
formulaReducedModel =  ~ sample + exon + cell_line:exon 
```

```{r}
dxd_nonrisk_Zn = DEXSeqDataSetFromHTSeq(
   countFiles,
   sampleData=sample_info,
   design= ~ User.ID + exon + condition:exon,
   flattenedfile=flattenedFile )
```

```{r}
dxd_nonrisk_Zn <- estimateSizeFactors(dxd_nonrisk_Zn)
dxd_nonrisk_Zn <- estimateDispersions( dxd_nonrisk_Zn, formula = formulaFullModel )
plotDispEsts(dxd_nonrisk_Zn)
```

```{r}
dxd_nonrisk_Zn = testForDEU( dxd_nonrisk_Zn, 
    reducedModel = formulaReducedModel, 
        fullModel = formulaFullModel )
```

```{r}
dxd_nonrisk_Zn <- estimateExonFoldChanges(dxd_nonrisk_Zn)
```

```{r}
dxr_nonrisk_Zn = DEXSeqResults( dxd_nonrisk_Zn )
```

MA plot:
```{r}
plotMA(dxr_nonrisk_Zn)
```

Collect significant exons/genes into a data frame:
```{r}
dxr_nonrisk_Zn_sig <- as.data.frame(dxr_nonrisk_Zn[dxr_nonrisk_Zn$padj < 0.05 & !is.na(dxr_nonrisk_Zn$padj),]) #padj <0.05
dxr_nonrisk_Zn_sig <- dxr_nonrisk_Zn_sig[dxr_nonrisk_Zn_sig$log2fold_treated_control < -0.58 | dxr_nonrisk_Zn_sig$log2fold_treated_control > 0.58,]  # -1<Log2FC<1
dxr_nonrisk_Zn_sig <- dxr_nonrisk_Zn_sig[dxr_nonrisk_Zn_sig$exonBaseMean>= 10,]
```

sort this data frame by adjusted p-value
```{r}
dxr_nonrisk_Zn_sig <- dxr_nonrisk_Zn_sig[order(dxr_nonrisk_Zn_sig$padj),]
```

```{r}
plotDEXSeq( dxr_nonrisk_Zn, "ENSG00000109339.18", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2, splicing = TRUE, expression = FALSE)
```
